google/gcloud/config.go

```diff
@@ -1,55 +1,36 @@
 package gcloud
 
+import (
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+)
+
 type Config struct {
-	ConfigDir    string        `json:"configDir" yaml:"configDir"`
-	Environments []Environment `json:"environments" yaml:"environments"`
+	ConfigPath string             `json:"configPath" yaml:"configPath"`
+	Accounts   map[string]Account `json:"accounts" yaml:"accounts"`
+	Clusters   map[string]Cluster `json:"clusters" yaml:"clusters"`
 }
 
-func (c Config) FindCluster(envName, clusterName string) (Cluster, bool) {
-	for _, env := range c.Environments {
-		if env.Name != envName {
-			continue
-		}
-
-		for _, cluster := range env.Clusters {
-			if cluster.Name == clusterName {
-				return cluster, true
-			}
-		}
+func (c Config) Cluster(name string) (Cluster, error) {
+	if value, ok := c.Clusters[name]; !ok {
+		return Cluster{}, errors.Errorf("given cluster not found: %s", name)
+	} else {
+		return value, nil
```

Comment:
else statement is not required in this form
https://go.dev/doc/effective_go#if
it's recommended to be omitted

Reply:
yes, I know it's "recommended" but in this case I preferring the variable scoped short declaration pattern